x-pack/plugin/shutdown/src/internalClusterTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownShardsIT.java

```diff
@@ -183,6 +186,202 @@ public void testNotStalledIfAllShardsHaveACopyOnAnotherNode() throws Exception {
+                        )
+                );
+            }, () -> fail("expected a 'NO' decision for nodeB but there was no explanation for that node"));
+    }
+
+    public void testNodeReplacementOverridesFilters() throws Exception {
+        String nodeA = internalCluster().startNode(Settings.builder().put("node.name", "node-a"));
+        // Create an index and pin it to nodeA, when we replace it with nodeB,
+        // it'll move the data, overridding the `_name` allocation filter
+        Settings.Builder nodeASettings = Settings.builder()
+            .put("index.routing.allocation.require._name", nodeA)
+            .put("index.number_of_shards", 3)
+            .put("index.number_of_replicas", 0);
+        createIndex("myindex", nodeASettings.build());
+        final String nodeAId = getNodeId(nodeA);
+        final String nodeB = "node_t1"; // TODO: fix this to so it's actually overrideable
+
+        // Mark the nodeA as being replaced
+        PutShutdownNodeAction.Request putShutdownRequest = new PutShutdownNodeAction.Request(
+            nodeAId,
+            SingleNodeShutdownMetadata.Type.REPLACE,
+            this.getTestName(),
+            null,
+            nodeB
+        );
+        AcknowledgedResponse putShutdownResponse = client().execute(PutShutdownNodeAction.INSTANCE, putShutdownRequest).get();
+        assertTrue(putShutdownResponse.isAcknowledged());
+
+        GetShutdownStatusAction.Response getResp = client().execute(
+            GetShutdownStatusAction.INSTANCE,
+            new GetShutdownStatusAction.Request(nodeAId)
+        ).get();
+
+        assertThat(getResp.getShutdownStatuses().get(0).migrationStatus().getStatus(), equalTo(STALLED));
+
+        internalCluster().startNode(Settings.builder().put("node.name", nodeB));
+        final String nodeBId = getNodeId(nodeB);
+
+        logger.info("--> NodeA: {} -- {}", nodeA, nodeAId);
+        logger.info("--> NodeB: {} -- {}", nodeB, nodeBId);
+
+        assertBusy(() -> {
+            ClusterState state = client().admin().cluster().prepareState().clear().setRoutingTable(true).get().getState();
+            for (ShardRouting sr : state.routingTable().allShards("myindex")) {
+                assertThat(
+                    "expected shard on nodeB (" + nodeBId + ") but it was on a different node",
+                    sr.currentNodeId(),
+                    equalTo(nodeBId)
+                );
+            }
+        });
+
+        assertBusy(() -> {
+            GetShutdownStatusAction.Response shutdownStatus = client().execute(
+                GetShutdownStatusAction.INSTANCE,
+                new GetShutdownStatusAction.Request(nodeAId)
+            ).get();
+            assertThat(shutdownStatus.getShutdownStatuses().get(0).migrationStatus().getStatus(), equalTo(COMPLETE));
+        });
+
+        final String nodeC = internalCluster().startNode();
```

Comment:
It would be sort of nice to start node C prior to the node B to ensure that the replace uses node B and not node C (which I think it might in the current state of this PR without `AllocationDeciders.canForceAllocateDuringReplace` implemented).